qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -547,8 +588,12 @@ def __eq__(self, other: "SymbolicPulse") -> bool:
         if self._envelope != other._envelope:
             return False
 
+        # _canonical_params is assumed to be a function of parameters. If parameters are the same,
+        # we don't need to check the _canonical_params. (Also solves the edge case of a pulse with
+        # no parameters)
         if self.parameters != other.parameters:
-            return False
+            if not self._equate_parameters(other):
```

Comment:
This is nitpicky, but the logic seems to me bit inefficient. It first fully evaluates the dict equality and then compares the canonicals and rest of dict items. Maybe just `return self._equate_paramters(other)` without if clause enough? Probably my suggestion is wrong because evaluation of builtin dict equality might be faster.